Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs

```diff
@@ -276,159 +282,66 @@ public async Task<IEnumerable<SubmissionForProfileServiceModel>> GetForProfileBy
     public async Task Submit(SubmitSubmissionServiceModel model)
     {
         var problem = await this.problemsDataService.GetWithProblemGroupCheckerAndTestsById(model.ProblemId);
-        if (problem == null)
-        {
-            throw new BusinessServiceException(Resources.ContestsGeneral.ProblemNotFound);
-        }
-
         var currentUser = this.userProviderService.GetCurrentUser();
-
         var participant = await this.participantsDataService
             .GetWithContestByContestByUserAndIsOfficial(
-                problem.ProblemGroup.ContestId,
+                problem!.ProblemGroup.ContestId,
```

Comment:
What if a problem is null? Add a null check before trying to call the problem property

Reply:
Done.